plugins/chat/assets/javascripts/discourse/components/chat-channel-row.gjs

```diff
@@ -196,7 +197,19 @@ export default class ChatChannelRow extends Component {
       >
         <ChannelIcon @channel={{@channel}} />
         <div class="chat-channel-row__info">
-          <ChannelName @channel={{@channel}} @unreadIndicator={{true}} />
+          <div class="chat-channel-row__name-container">
+            <ChannelName @channel={{@channel}} @unreadIndicator={{true}} />
+            {{#if
+              (and
+                @channel.currentUserMembership.starred
+                this.siteSettings.star_chat_channels
+              )
+            }}
```

Comment:
I think in this case you can just check `currentUserMembership.starred` and not the site setting, since `starred` will never be true if the site setting is false because of your serializer